### PR TITLE
chore(views): Remove all deprecated views

### PR DESCRIPTION
**Pull request by ewinslow**
BREAKING CHANGE:
Several views that were deprecated in the 1.x series are being
removed in 2.x. This helps us keep the codebase clean and our
maintenance burden lower.

### CODE REVIEW

File: views/default/page/elements/sidebar.php [hunk 1]
```diff
@@ -18,15 +18,3 @@
 if (isset($vars['sidebar'])) {
 	echo $vars['sidebar'];
 }
-
-// @todo deprecated so remove in Elgg 2.0
-// optional second parameter of elgg_view_layout
-if (isset($vars['area2'])) {
-	echo $vars['area2'];
```
Comment (mrclay) [inline views/default/page/elements/sidebar.php]: Where do devs get indication that this was deprecated?

Comment (mrclay) [inline views/default/page/elements/sidebar.php]: Others LGTM

**Comment by mrclay** [comment]
Seems should mention in docs/guides/upgrading

**Comment by ewinslow** [comment]
Added pretty thorough docs on what was dropped. Merging this in now.

**Comment by juho-jaakkola** [comment]
Wait for at least two LGTMs before merging breaking changes, please.

**Comment by ewinslow** [comment]
Apologies I did not think this would be controversial as these deprecations are all pretty old and trivial to support again via plugin.

**Comment by juho-jaakkola** [comment]
No worries about the PR itself. My comment was more about the principle than this particular PR.

**Comment by ewinslow** [comment]
Sounds good.

**Pull request merged**